Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -28,6 +28,18 @@ class APIClient {
         return operationQueue
     }()
 
+    /// Queue in charge of handling recovery related requests. Handles operations in serial
+    private let recoveryQueue: OperationQueue = {
+        let operationQueue = OperationQueue()
+        operationQueue.name = "com.stream.api-client.recovery"
+        operationQueue.maxConcurrentOperationCount = 1
+        return operationQueue
+    }()
+
+    /// Determines whether the APIClient is in recovery mode. During recovery period we limit the concurrent operations to 1, and we only allow recovery related
+    /// requests to be run,
+    @Atomic private var isRecoveryModeOn: Bool = false
```

Comment:
Something like
```suggestion
    @Atomic private var isInRecoveryMode: Bool = false
```

might be more readable. Furtherly, does it makes sense to turn this property into readonly?
```
private (set) var ...
```